glTFRevitExport/Utils/GeometryUtils.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using Vector3D = System.Numerics.Vector3;
+
+namespace GLTFRevitExport
+{
+    public class GeometryUtils
+    {
+        public static bool BBoxCurveIntersection(Vector3D B1, Vector3D B2, Vector3D L1, Vector3D L2, ref Vector3D Hit)
+        {
+            if (L2.X < B1.X && L1.X < B1.X) return false;
+            if (L2.X > B2.X && L1.X > B2.X) return false;
+            if (L2.Y < B1.Y && L1.Y < B1.Y) return false;
+            if (L2.Y > B2.Y && L1.Y > B2.Y) return false;
+            if (L2.Z < B1.Z && L1.Z < B1.Z) return false;
+            if (L2.Z > B2.Z && L1.Z > B2.Z) return false;
+            if (L1.X > B1.X && L1.X < B2.X &&
+                L1.Y > B1.Y && L1.Y < B2.Y &&
+                L1.Z > B1.Z && L1.Z < B2.Z)
+            {
+                Hit = L1;
+                return true;
+            }
+            if ((GetIntersection(L1.X - B1.X, L2.X - B1.X, L1, L2, ref Hit) && InBox(Hit, B1, B2, 1))
+              || (GetIntersection(L1.Y - B1.Y, L2.Y - B1.Y, L1, L2, ref Hit) && InBox(Hit, B1, B2, 2))
+              || (GetIntersection(L1.Z - B1.Z, L2.Z - B1.Z, L1, L2, ref Hit) && InBox(Hit, B1, B2, 3))
+              || (GetIntersection(L1.X - B2.X, L2.X - B2.X, L1, L2, ref Hit) && InBox(Hit, B1, B2, 1))
+              || (GetIntersection(L1.Y - B2.Y, L2.Y - B2.Y, L1, L2, ref Hit) && InBox(Hit, B1, B2, 2))
+              || (GetIntersection(L1.Z - B2.Z, L2.Z - B2.Z, L1, L2, ref Hit) && InBox(Hit, B1, B2, 3)))
+                return true;
+
+            return false;
+        }
+
+        private static bool GetIntersection(float fDst1, float fDst2, Vector3D P1, Vector3D P2, ref Vector3D Hit)
+        {
+            if ((fDst1 * fDst2) >= 0.0f) return false;
+            if (fDst1 == fDst2) return false;
+            Hit = P1 + (P2 - P1) * (-fDst1 / (fDst2 - fDst1));
+            return true;
+        }
+
+        private static bool InBox(Vector3D Hit, Vector3D B1, Vector3D B2, int Axis)
+        {
+            if (Axis == 1 && Hit.Z > B1.Z && Hit.Z < B2.Z && Hit.Y > B1.Y && Hit.Y < B2.Y) return true;
```

Comment:
try to put the hardcodes like 1, 2, 3 in constant with representatives names

Reply:
Great point! I've implemented it!